components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -8154,27 +8202,48 @@ public boolean editComment(ApiTypeWrapper apiTypeWrapper, String commentId, Comm
      * @throws APIManagementException
      */
     public boolean deleteComment(ApiTypeWrapper apiTypeWrapper, String commentId) throws APIManagementException {
-        try (Connection connection = APIMgtDBUtil.getConnection()) {
-            int id = -1;
-            String deleteCommentQuery = SQLConstants.DELETE_COMMENT_SQL;
+        try (Connection connection = APIMgtDBUtil.getConnection()){
             Identifier identifier;
             if (apiTypeWrapper.isAPIProduct()) {
                 identifier = apiTypeWrapper.getApiProduct().getId();
             } else {
                 identifier = apiTypeWrapper.getApi().getId();
             }
+            return deleteComment(identifier, commentId, connection);
+        } catch (SQLException e) {
+            handleException("Error while deleting comment " + commentId + " from the database", e);
+        }
+        return false;
+    }
+
+    private boolean deleteComment(Identifier identifier, String commentId, Connection connection) throws APIManagementException {
+        int id = -1;
+        String deleteCommentQuery = SQLConstants.DELETE_COMMENT_SQL;
+        String getCommentIDsOfReplies = SQLConstants.GET_IDS_OF_REPLIES_SQL;
+        ResultSet resultSet = null;
+        try {
             id = getAPIID(identifier, connection);
             if (id == -1) {
                 String msg = "Could not load API record for: " + identifier.getName();
                 throw new APIManagementException(msg);
```

Comment:
Let's use an exception code.
```suggestion
                throw new APIManagementException(msg, ExceptionCodes.from(ExceptionCodes.API_NOT_FOUND, existingApiId));
```